services/wallet/transfer/transaction.go

```diff
@@ -78,231 +87,110 @@ type MultiTransactionCommandResult struct {
-		&transaction.Data,
-		&transaction.Symbol,
-		(*bigint.SQLBigIntBytes)(transaction.GasPrice.Int),
-		(*bigint.SQLBigIntBytes)(transaction.GasLimit.Int),
-		&transaction.Type,
-		&transaction.AdditionalData,
-		&transaction.ChainID,
-		&transaction.MultiTransactionID,
-	)
+	hashes, err := tm.sendTransactions(multiTransaction, data, bridges, password, rpcClient)
 	if err != nil {
 		return nil, err
 	}
 
-	return transaction, nil
-}
-
-func (tm *TransactionManager) AddPending(transaction PendingTransaction) error {
-	insert, err := tm.db.Prepare(`INSERT OR REPLACE INTO pending_transactions
-                                      (network_id, hash, timestamp, value, from_address, to_address,
-                                       data, symbol, gas_price, gas_limit, type, additional_data, multi_transaction_id)
-                                      VALUES
-                                      (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`)
+	err = tm.storePendingTransactions(multiTransactionID, multiTransaction, hashes, data)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	_, err = insert.Exec(
-		transaction.ChainID,
-		transaction.Hash,
-		transaction.Timestamp,
-		(*bigint.SQLBigIntBytes)(transaction.Value.Int),
-		transaction.From,
-		transaction.To,
-		transaction.Data,
-		transaction.Symbol,
-		(*bigint.SQLBigIntBytes)(transaction.GasPrice.Int),
-		(*bigint.SQLBigIntBytes)(transaction.GasLimit.Int),
-		transaction.Type,
-		transaction.AdditionalData,
-		transaction.MultiTransactionID,
-	)
-	return err
-}
 
-func (tm *TransactionManager) DeletePending(chainID uint64, hash common.Hash) error {
-	_, err := tm.db.Exec(`DELETE FROM pending_transactions WHERE network_id = ? AND hash = ?`, chainID, hash)
-	return err
+	return &MultiTransactionCommandResult{
+		ID:     int64(multiTransactionID),
+		Hashes: hashes,
+	}, nil
 }
 
-func (tm *TransactionManager) Watch(ctx context.Context, transactionHash common.Hash, client *chain.ClientWithFallback) error {
-	watchTxCommand := &watchTransactionCommand{
-		hash:   transactionHash,
-		client: client,
-	}
-
-	commandContext, cancel := context.WithTimeout(ctx, 10*time.Minute)
-	defer cancel()
+func (tm *TransactionManager) storePendingTransactions(multiTransactionID MultiTransactionIDType,
+	multiTransaction *MultiTransaction, hashes map[uint64][]types.Hash, data []*bridge.TransactionBridge) error {
 
-	return watchTxCommand.Command()(commandContext)
+	txs := createPendingTransactions(hashes, data, multiTransactionID, multiTransaction)
+	for _, tx := range txs {
+		err := tm.pendingManager.AddPending(tx)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-const multiTransactionColumns = "from_address, from_asset, from_amount, to_address, to_asset, to_amount, type, timestamp"
+func createPendingTransactions(hashes map[uint64][]types.Hash, data []*bridge.TransactionBridge,
+	multiTransactionID MultiTransactionIDType, multiTransaction *MultiTransaction) []*transactions.PendingTransaction {
```

Comment:
no need to pass `multiTransactionID` separately, it's part of the `multiTransaction` object

Reply:
It didn't before, but now it gets set when you insert the item into the DB